src/test/java/software/sigma/internship/service/TeacherServiceTest.java

```diff
@@ -24,34 +23,81 @@ public class TeacherServiceTest {
     @Mock
     private TeacherRepository teacherRepository;
 
-    @Spy
-    private TeacherMapper teacherMapper;
+    @Mock
+    private ModelMapper teacherMapper;
 
     @InjectMocks
     private TeacherServiceImpl teacherService;
 
     private final Teacher teacher = new Teacher();
+    private final TeacherDto teacher1Dto = new TeacherDto();
+    private final Teacher newSavedTeacher = new Teacher();
+    private final TeacherDto teacherDto = new TeacherDto();
+    private final TeacherDto teacherEntityToDto = new TeacherDto();
+    private final Teacher teacherDtoToEntity = new Teacher();
 
     @Before
     public void setup() {
         teacher.setPosition(Teacher.Position.DOCENT);
         teacher.setId(1L);
         teacher.setFirstName("Ivan");
         teacher.setLastName("Ivanov");
+
+        teacher1Dto.setPosition(Teacher.Position.DOCENT);
+        teacher1Dto.setId(1L);
+        teacher1Dto.setFirstName("Ivan");
```

Comment:
Why do you need two different variables teacher1Dto and teacherDto? Is it possible to use only one? 
Also, it would be better setuping teacher move to the external method.

Reply:
`teacher` is an Entity (with id)
`teacher1Dto` is a DTO which is being returned by `tacherMapper` after converting `teacher` (also with id)
`teacherDto` is a DTO without id. I save it to check if new id is assigned to it.

Reply:
Also, what do you mean by _external method_?

Reply:
As I understand teacher with Id means that teacher is already saved and without id means that teacher is new. 
Maybe it would be better to rename savedTeacher and newTeacher or something like this?

Reply:
Agree, names should make sense.